tests/test_pooling.py

```
@@ -0,0 +1,39 @@
+
+    def test_forward_kernel3_stride1(self):
+        x = np.random.randn(1, 2, 6, 6).astype(np.float32)
+
+        y = funcs.max_pooling_2d(x, kernel_size=(3, 3), stride=1)
+        expected = F.max_pool2d(torch.tensor(x), kernel_size=3, stride=1)
+
+        self.assertTrue(array_close(y.data, expected.numpy()))
+
+    def test_backward(self):
+        x = np.random.randn(2, 2, 6, 6)
+        f = lambda x_in: funcs.max_pooling_2d(x_in, kernel_size=(2, 2), stride=2)
```

Ruff (0.15.15)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `f` as a `def`

(E731)